src/mbgl/storage/resource.cpp

```diff
@@ -123,7 +123,7 @@ Resource Resource::tile(const std::string& urlTemplate,
             } else if (token == "bbox-epsg-3857") {
                 return getTileBBox(x, y, z);
             } else if (token == "prefix") {
-                std::string prefix{ 2 };
+                std::string prefix{ "00" };
```

Comment:
Ooof, good catch. It looks like this is a pretty bad trap in the `std::string` constructor. Instead of choosing the `string(size_t, char)` constructor that I intended, it implicitly converted `2` to a `std::initializer_list<char>` and chose `string(std::initializer_list<char>)`. This was only possibly because I was using list initialization with braces, which chooses the `initializer_list` constructor in this case. This wouldn't have happened with `std::string prefix(2)` (compiler error). Thanks for catching this! I assume our memory sanitizer didn't catch this because we weren't _technically_ writing out of bounds; just overwriting the null terminator in the string.